docs/bazel/status.md

```diff
@@ -0,0 +1,339 @@
+  * [Go](#tasks_go)
+  * [Rust](#tasks_rust)
+  * [Javascript](*tasks_js)
+  * [Integration](#tasks_integration)
+* [Status](#status)
+* [Schedule](#schedule)
+* [Working with Bazel](#workingwithbazel)
+* [Resources](#resources)
+
+## Goals and Non-goals <a name="goals"></a>
+
+Mandatory Goals:
+
+* Complete Bazel coverage of the Coda codebase:
+  * OCaml code
+  * C/C++ code and dependencies.
+  * Golang dependencies ([go-libp2p](https://github.com/libp2p/go-libp2p))
+  * Rust dependencies ([marlin](https://github.com/scipr-lab/marlin)?)
+  * Reason/Javascript
+* "Just works" on:
+  * Linux
+  * MacOS
+
+Non-mandatory goals:
+
+* Cross-platform builds targeting Linux, x86_64
+  * In particular, MacOS -> Linux
+* Dockerfile to support Linux development of Coda on Mac host
+
+Non-goals:
+
+* Windows support.  Maybe later.
+* Cross-platform build targeting ARM.
+* Production-quality `rules_ocaml` Bazel library.  The goal is
+  implement just enough functionality to build Coda.  Supporting a
+  production-quality general-purpose library of OCaml rules for Bazel
+  is out of scope.
+
+## Tasks <a name="tasks"></a>
+
+General:
+
+* Customization/Optimiztion configurations.
+  * User-friendly support for settings e.g. debug/optimized build, etc.
+  * Support for dependency configuration.  For example, libsnark
+    supports various curves (bn128, etc.), building with or without
+    procps support, etc.
+
+### C/C++ Tasks <a name="tasks_c"></a>
+
+#### Quick Links:
+* [Developer readme](README-dev.md)
+* [Compiling from source and running a node](docs/demo.md)
+* [Directory structure](frontend/website/docs/developers/directory-structure.md)
+* [Lifecycle of a payment](frontend/website/docs/architecture/lifecycle-payment.md)
+
+* Take inventory of all C/C++ dependencies, both internal and external.
+* Write Bazel code for all C/C++ dependencies.
+  * For "native" builds: build host == execution host == target host
+  * Cross-platform builds: build/exec host == Linux, MacOS; target host == Linux x86_64
+
+[libsnark](https://github.com/scipr-lab/libsnark) gets special treatment, since it is central to Coda.
+Coda vendors this library - embeds a copy of it within the Coda
+codebase
+([src/lib/snarky/src/camlsnark_c/libsnark-caml/libsnark](src/lib/snarky/src/camlsnark_c/libsnark-caml/libsnark)),
+with some local modifications.
+
+Several strategies for Bazelization of libsnark are available:
+
+* Add direct (embedded) Bazel support to libsnark and all or some of its dependencies; this entails one of:
+  * Convince the upstream maintainers to incorporate Bazel support; or
+  * Maintain the Bazelized forks in the Coda repo; or
+  * Incorporate the Bazel code into the vendored image of libsnark within Coda.
+* Import libsnark as an external repo and write a local BUILD file for it.
+
+This implies a choice:
+
+* Retain the current vendored image and add Bazel support to it.
+* Treat libsnark as an external repo, and patch/augment it for Coda purposes
```

Comment:
I believe I've got bazelized forks of libsnark and all its deps.  It would probably be wise for o1-labs to take custody of those forks at some point.  Plan is to submit PRs upstream, but who knows if/when they would be accepted.